File: dx-api/app/consts/scoring.go
package consts

// Scoring consts for game score calculation.
const (
	CorrectAnswer        = 1
	Combo3Bonus          = 3
	Combo5Bonus          = 5
	Combo10Bonus         = 10
	ComboCycleLength     = 10
	LevelCompleteExp     = 10
	ExpAccuracyThreshold = 0.6
)

// ComboThreshold defines the streak count and its associated bonus.
type ComboThreshold struct {
	Streak int
	Bonus  int
}

// ComboThresholds lists the combo bonus tiers in ascending order.
var ComboThresholds = []ComboThreshold{
	{Streak: 3, Bonus: Combo3Bonus},
	{Streak: 5, Bonus: Combo5Bonus},
	{Streak: 10, Bonus: Combo10Bonus},
}

File: dx-web/src/consts/scoring.ts
export const SCORING = {
  CORRECT_ANSWER: 1,
  COMBO_3_BONUS: 3,
  COMBO_5_BONUS: 5,
  COMBO_10_BONUS: 10,
  COMBO_CYCLE_LENGTH: 10,
  LEVEL_COMPLETE_EXP: 10,
  EXP_ACCURACY_THRESHOLD: 0.6,
} as const;

export const COMBO_THRESHOLDS = [
  { streak: 3, bonus: SCORING.COMBO_3_BONUS },
  { streak: 5, bonus: SCORING.COMBO_5_BONUS },
  { streak: 10, bonus: SCORING.COMBO_10_BONUS },
] as const;
